### PR TITLE
Fixed CustomLogger super call

### PR DESCRIPTION
**Pull request by mattp-**
In cases where another logger uses getLoggerClass to subclass and then
setLoggerClass's itself,  molecules logger passes the wrong value to
super() causing a recursionError.

Please include details of what it is, how to use it, how it's been tested

#### PR Type
- Bugfix Pull Request

### CODE REVIEW

**Comment by ssbarnea** [comment]
@mattp- please rebase, we fixed the unrelated CI failures.

File: molecule/test/unit/test_logger.py [hunk 1]
```diff
@@ -177,3 +179,16 @@ def test_markup_detection_tty_no(mocker):
...
+    class FooLogger(logging.getLoggerClass()):
+        log = logging.getLogger('FooLogger')
+        pass
+
+    logging.setLoggerClass(FooLogger)
+
+    try:
+        log = FooLogger('foo')
+    except RecursionError as exc:
+        raise pytest.fail("CustomLogger caused recursion error: {0}".format(exc))
```
Comment (greg-hellings) [inline molecule/test/unit/test_logger.py:194]: Why are you catching this, and not just letting the error pass out to pytest automatically?

Comment (mattp-) [reply molecule/test/unit/test_logger.py:194]: not sure, I guess that also works :) ill change

**Comment by greg-hellings** [comment]
You accidentally made an extra test and it passes. Somewhere in the test code you'll see definitions for how many tests there are supposed to be. You'll need to bump that up to 433 to account for your additional test.

**Comment by mattp-** [comment]
@greg-hellings should be good now I believe, pending that last build-containers task. thanks!

**Review by greg-hellings — APPROVED** [review]
lgtm

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**